bios/stage-4/src/main.rs

```diff
@@ -56,13 +56,21 @@ pub extern "C" fn _start(info: &mut BiosInfo) -> ! {
         PhysAddr::new(info.kernel.start)
     };
     let kernel_size = info.kernel.len;
-    let mut frame_allocator = {
+    let mut frame_allocator = if info.ramdisk.start == 0 {
```

Comment:
Right now we always set the `start` field in `stage-2/src/main.rs`, so this condition is never true. We should check the `ramdisk_len` instead.